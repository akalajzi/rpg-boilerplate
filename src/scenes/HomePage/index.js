// @flow strict
import React from 'react';
import styled from 'styled-components';

import defaultTheme from 'themes/default';

import hiCat from 'static/images/hi.gif';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding-top: 40px;
`;

const WelcomeText = styled.div`
  align-items: center;
  justify-content: center;
  font-size: 24px;
  font-weight: 700;
  color: ${({ theme }: ThemeProps) => theme.colors.orange};
`;

const HiImage = styled.img`
  margin-top: 20px;
`;

WelcomeText.defaultProps = {
  theme: defaultTheme,
};

const HomePage = () => (
  <Container>
    <WelcomeText>Oh hai</WelcomeText>
    <HiImage src={hiCat} alt="Hi" />
  </Container>
);

export default HomePage;
