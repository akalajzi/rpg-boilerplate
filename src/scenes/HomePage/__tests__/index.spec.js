// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';

import HomePage from '../index';

describe('#HomePage', () => {
  test('render', () => {
    const component = renderer.create(<HomePage />);

    expect(component).toMatchSnapshot();
  });
});
