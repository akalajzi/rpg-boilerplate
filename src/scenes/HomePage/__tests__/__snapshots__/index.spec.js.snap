// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#HomePage render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-top: 40px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 24px;
  font-weight: 700;
  color: #FFA500;
}

.c2 {
  margin-top: 20px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Oh hai
  </div>
  <img
    alt="Hi"
    className="c2"
    src="test-file-stub"
  />
</div>
`;
